dialogy/plugins/text/oos_filter/__init__.py

```diff
@@ -0,0 +1,34 @@
+"""
+.. OOSFilter: Mutate an incoming intent to out-of-scope intent (OOS) if the score of the predicted intent is less than certain threshold
+
+Expects 2 mandatory arguments:
+1. intent_oos -> str
+2. threshold value -> float
+"""
+
+from typing import Any, Callable, Dict, List, Optional
+
+from dialogy.base.plugin import Input, Output, Plugin
+from dialogy.types import Intent
+
+
+class OOSFilterPlugin(Plugin):
+    def __init__(
+        self,
+        dest=None,
+        guards=None,
+        threshold: float = 0.5,
+        intent_oos: str = "_oos_",
+        **kwargs
+    ) -> None:
+        super().__init__(dest=dest, guards=guards, **kwargs)
+        self.threshold = threshold
+        self.intent_oos = intent_oos
+
+    def set_oos_intent(self, intents: List[Intent]) -> Any:
+        if intents[0].score < self.threshold:
```

Comment:
Before accessing an index of the list, always check if the list is empty or not.
In this case, we should raise an exception if the list is empty and the plugin was invoked, since it would mean that `OOSFilterPlugin` was added in the pipeline before the intent classifier which is invalid. Please add that exception handling and add a unit test for it.